src/fsharp/DependencyManager.Integration.fs

```diff
@@ -0,0 +1,230 @@
+    |> Seq.choose (fun path -> try Assembly.LoadFrom path |> Some with | _ -> None)
+    |> Seq.filter (fun a -> ReflectionHelper.assemblyHasAttribute a "FSharpDependencyManagerAttribute")
+
+type ProjectDependencyManager() =
+    interface IDependencyManagerProvider with
+        member __.Name = "Project loader"
+        member __.ToolName = ""
+        member __.Key = "project"
+        member __.ResolveDependencies(_targetFramework:string, _scriptDir: string, _scriptName: string, _packageManagerTextLines: string seq) = 
+            None,[]
+
+    interface System.IDisposable with
+        member __.Dispose() = ()
+
+type RefDependencyManager() =
+    interface IDependencyManagerProvider with
+        member __.Name = "Ref library loader"
+        member __.ToolName = ""
+        member __.Key = "ref"
+        member __.ResolveDependencies(_targetFramework:string, _scriptDir: string, _scriptName: string, _packageManagerTextLines: string seq) = 
+            None,[]
+
+    interface System.IDisposable with
+        member __.Dispose() = ()
+
+type ImplDependencyManager() =
+    interface IDependencyManagerProvider with
+        member __.Name = "Impl library loader"
+        member __.ToolName = ""
+        member __.Key = "impl"
+        member __.ResolveDependencies(_targetFramework:string, _scriptDir: string, _scriptName: string, _packageManagerTextLines: string seq) = 
+            None,[]
+
+    interface System.IDisposable with
+        member __.Dispose() = ()
+
+let registeredDependencyManagers = lazy (
+    let defaultProviders =
+        [new ProjectDependencyManager() :> IDependencyManagerProvider
+         new RefDependencyManager() :> IDependencyManagerProvider
+         new ImplDependencyManager() :> IDependencyManagerProvider]
+    
+    let loadedProviders =
+        enumerateDependencyManagerAssembliesFromCurrentAssemblyLocation()
+        |> Seq.collect (fun a -> a.GetTypes())
+        |> Seq.choose ReflectionDependencyManagerProvider.InstanceMaker
+        |> Seq.map (fun maker -> maker ())
+    
+    defaultProviders
+    |> Seq.append loadedProviders
+    |> Seq.map (fun pm -> pm.Key, pm)
+    |> Map.ofSeq
+)
+
+let RegisteredDependencyManagers() = registeredDependencyManagers.Force()
+
+let createPackageManagerUnknownError packageManagerKey m =
+    let registeredKeys = String.Join(", ", RegisteredDependencyManagers() |> Seq.map (fun kv -> kv.Value.Key))
+    let searchPaths = assemblySearchPaths.Force()
+    Error(FSComp.SR.packageManagerUnknown(packageManagerKey, String.Join(", ", searchPaths), registeredKeys),m)
+
+let tryFindDependencyManagerInPath m (path:string) : ReferenceType =
+    try
+        match path.IndexOf(":") with
+        | -1 | 1 ->
+            ReferenceType.Library path
+        | _ ->
+            let managers = RegisteredDependencyManagers()
+            match managers |> Seq.tryFind (fun kv -> path.StartsWith(kv.Value.Key + ":" )) with
```

Comment:
We need to fix the lookup, didn't had chance to fix that in original PR but we are using a map, so we should lookup on the exact key rather than iteration.